editor/editor_node.cpp

```diff
@@ -6799,6 +6799,7 @@ EditorNode::EditorNode() {
 
 EditorNode::~EditorNode() {
 	EditorInspector::cleanup_plugins();
+	EditorTranslationParser::get_singleton()->clean_parsers();
```

Comment:
Can't this be done in `EditorTranslationParser`'s destructor directly?

Reply:
When I'm debugging, I couldn't detect the destructor of `EditorTranslationParser` being called - `print_line()` doesn't print. I assume the destructor is not called or only called when it's too late (after the leak is detected), because of how I create the singleton.

So I tried making `EditorTranslationParser` inherit from `Node`, then do `singleton = memnew(EditorTranslationParser)` and `add_child()` in EditorNode constructor. See [example ](https://github.com/godotengine/godot/blob/master/editor/editor_node.cpp#L5705). The idea is to let `EditorNode` clear its children together with `EditorTranslationParser`, so that the destructor will be called. Unfortunately `GDScriptTranslationParser` wants to register first before we enter `EditorNode` constructor, so the singleton will be empty and it will crash. 

I consider some other ways too. But it wasn't elegant :/